System / simulation implemenation for first two scenarios

willemvanoort opened the pull request:
All,

Please review the code changes. Long/lat coordinate system for the implementation might overcomplicate things, but I have gotten the scenarios to work as expected. Any feedback/thoughts/improvements/suggestions? Thanks!

willemvanoort committed:
Implementation for first two scenarios (utest tbd)

willemvanoort committed:
Cleanup of system/simulation docstrings

JFoederer commented on demo/Titanic/simulation/floating_object.py:
```diff
@@ -1,30 +1,80 @@
+from __future__ import annotations
+
 import math
```
We will have to see how to deal with new dependencies. I do not want to expand the dependencies of the core library due to the demo, but I also want people to easily run the demo directly from the archive.

JFoederer replied:
In the easy to run area, we also have to think about the python path dependency. (Not part of this PR)

willemvanoort replied:
Ah yea, will remove this one by simply removing some annotations (does not affect anything at runtime anyways)

JFoederer replied:
If they are build-in libraries then it doesn't have to be an issue. I do think we have to prefer simplicity over accuracy to stay in line with the requirement of extendability. We want to keep it suitable for use in a one day workshop.

JFoederer replied:
I already included an _Install dependencies_ step in the installation instructions, even though for now this is just the robotMBT library. We can add a `requirements.txt` file with a pip instruction for additional dependencies when they arrive. For instance when adding the gamification branch.

JFoederer commented on demo/Titanic/simulation/titanic_in_ocean.py:
```diff
@@ -1,17 +1,41 @@
+    def update_coordinates(self):
+        # TODO Take into account the steering_direction of the titanic
+        # TODO Take into account the throttle of the titanic (i.e.: should start decreasing speed when throttle is 0)
+        return super().update_coordinates()
```
Test run fails here due to missing argument definition compared to the base class. I think the method can be omitted in full.

willemvanoort replied:
It can be, but i wanted to keep the todos in there such that they can be picked up. Fixed now

JFoederer replied:
Good point, we will need that.

JFoederer commented on demo/Titanic/domain_lib/JourneyLib.py:
```diff
@@ -1,30 +1,52 @@
-    def pass_time(self):
-        j = Journey(self.ocean)
-        j.passed_time(10)
+    @keyword("play out Journey for a duration of ${minutes} minutes")
```
Logging is very limited from this keyword. I think we need a bit more transparency to follow along what is happening.

willemvanoort replied:
Agreed. Will for now add some statements about what is going on with the Titanic.

BaharKocamanoglu commented on demo/Titanic/domain_lib/JourneyLib.py:
```diff
@@ -1,30 +1,52 @@
-    def spawn(self, n:int , w:int, heading:int, speed:int):
-        t = Titanic(speed, "east")
-        tio = TitanicInOcean(t, n, w, speed, "west")
+    @keyword("Spawn titanic at coordinate long ${n} lat ${w} heading ${heading} at speed of ${speed} knots")
```
I would prefer to use longitude and latitude instead of long and lat

willemvanoort replied:
Will update that

willemvanoort committed:
Fix update_coordinates method for TitanicInOcean

JFoederer committed:
Build top-down story structure (scenarios only)

JFoederer committed:
Add model-info for story structure

Empty step implementations are added that have some (preliminary) model-info included to run the story in the correct order. It will pass when selecting the happy path and fail when hitting the iceberg.

JFoederer committed:
Unify port-to-port sailing on leg and voyage levels

willemvanoort committed:
Merge branch 'Story-structure' into topic-enhanced-titanic-demo

BaharKocamanoglu committed:
Updated the library and resource definition

willemvanoort committed:
Initial keyword implementation (untested)

dsm-topic committed:
Refine Itinerary Scenario with 'in the port of'

Commit:
Add MapLib kw

foxinboxxxx committed:
Add MapLib kw

BaharKocamanoglu committed:
Keyword implementation

Keyword implementation for robot files

BaharKocamanoglu committed:
Merge branch 'topic-enhanced-titanic-demo' of https://github.com/foxinboxxxx/robotframeworkMBT into topic-enhanced-titanic-demo

willemvanoort committed:
Full suite working with SUT now

willemvanoort committed:
Hits iceberg scenario implementation + check arrival date implementation

JFoederer committed:
Use datetime from standard lib

JFoederer committed:
Fail without a PASS on the Iceberg alley section

dsm-topic committed:
step_defs.resource_orig: remove file

dsm-topic committed:
sinks -> sink

JFoederer committed:
Update position-on-map detection

JFoederer committed:
Remove the need for the departure port

BaharKocamanoglu committed:
Some log lines are added

JFoederer committed:
Log coordinate string rather than object data

JFoederer committed:
Assume an unsinkable ship (continue after the hit)

willemvanoort committed:
Update collision threshold to add support for barely missing the iceberg

JFoederer committed:
Moved some stuff around for presentation purposes

JFoederer committed:
Clean up imports

dsm-topic committed:
Add feature file & sync tests to feature file

dsm-topic committed:
Fastest_service.md: fix case

dsm-topic committed:
Atlantic_crossing.md: remove unneeded file

dsm-topic committed:
Fastest_service.md: move example heading

JFoederer committed:
Link to external image

JoFoPro committed:
Fix small logging mistakes

JFoederer committed:
Remove unimplemented scenario

For demo purposes it is clearer when this scenario is not yet here.

JFoederer committed:
Remove unused files

JFoederer committed:
Add run script for the demo

JFoederer committed:
Add documentation to Titanic demo

JFoederer committed:
Undo .gitignore change
